### PR TITLE
chore: fix CI and refactor patch deployment script

### PR DESCRIPTION
**Pull request by yomarion**
Bandaid for #20

fix: CI with `rustup default 1.67.0` quick-win
chore: Github workflow should only run on PR targeting master or push to master, not on every push
chore: Github workflow includes linting checks (+ checks applied)
chore: refactor patch deployment script
chore: fix error log when one transfer goes wrong

### CODE REVIEW

File: fungible_proxy/src/lib.rs [hunk 1]
```diff
@@ -181,7 +181,7 @@ impl FungibleProxy {
         } else {
              // return full amount for `ft_resolve_transfer` on the token contract
             let change = (amount.0 + args.fee_amount.0).to_string();
-            log!("Failed to transfer to account {}. Returning attached amount of {} of token {} to {}", args.to, change, token_address, payer);
+            log!("Transfer failed to {} or {}. Returning attached amount of {} of token {} to {}", args.to, args.fee_address, change, token_address, payer);
```
Comment (yomarion) [inline fungible_proxy/src/lib.rs]: We don't know which transfer failed. We could but the token will probably throw an explicit error anyway.

File: deploy.sh [hunk 2]
```diff
@@ -20,21 +21,28 @@ while getopts "pha:-:" OPT; do
   fi
   case "$OPT" in
     h | help)
-      echo "Builds and deploys the contract with state initialization (first deployement)."
+      echo "Builds and deploys contracts, eventually with state initialization."
```
Comment (MantisClone) [inline deploy.sh]: "eventually" seems like an unnecessary word here.

```suggestion
      echo "Builds and deploys contracts with state initialization."
```

File: deploy.sh [hunk 3]
```diff
@@ -20,21 +21,28 @@ while getopts "pha:-:" OPT; do
...
       echo "Defaults to testnet."
       echo ""
       echo "Options:"
+      echo "  -h | --help                 : shows this help"
       echo "  -p | --prod | --mainnet     : for prod deployment"
       echo "  -a [account_id]             : to override \$ACCOUNT_ID"
+      echo "  --patch                     : to patch an existing contract (skip the init function if any)"
```
Comment (MantisClone) [inline deploy.sh]:
```suggestion
      echo "  --patch                     : to patch an existing contract (skip the init function, if any)"
```

File: deploy.sh [hunk 4]
```diff
@@ -45,16 +53,26 @@ if [ "$ACCOUNT_ID" = "" ]; then
...
 fi
 
-printf "Deploying %s on NEAR_ENV=%s with ACCOUNT_ID=%s\n\n" "$contract_name" "$NEAR_ENV" "$ACCOUNT_ID"
+printf "Deploying %s on NEAR_ENV=%s with ACCOUNT_ID=%s (patch=%s)\n\n" "$contract_name" "$NEAR_ENV" "$ACCOUNT_ID" "$patch"
+
 
 ./build.sh
 
 if [ "$contract_name" = "fungible_proxy" ]; then
+  set -x
```
Comment (MantisClone) [inline deploy.sh:62]: Is this for debugging? Do you want to keep it in there?

**Review by MantisClone — CHANGES_REQUESTED** [review]

File: deploy.sh (hunk 4 above)
Comment (yomarion) [reply deploy.sh:62]: Yes, I plan to keep it here to show the commands that are run, so it's easier to understand what's going on, especially if something went wrong, but not only.

File: tests/sim/conversion_proxy.rs [hunk 5]
```diff
@@ -94,7 +94,6 @@ fn test_transfer() {
         ),
         deposit = transfer_amount
     );
-    result.assert_success();
```
Comment (MantisClone) [inline tests/sim/conversion_proxy.rs]: Why is this line removed?

File: .github/workflows/build-and-test.yml [hunk 6]
```diff
@@ -1,34 +1,40 @@
-name: build and test
+name: main
```
Comment (MantisClone) [inline .github/workflows/build-and-test.yml:1]: Consider renaming the workflow to match.

File: .github/workflows/build-and-test.yml [hunk 7]
```diff
@@ -1,34 +1,40 @@
...
 
 on:
   push:
+    branches:
+      - "main" 
   pull_request:
-    branches: [ "main" ]
+    branches:
+      - "main" 
+    types: [opened, synchronize]
```
Comment (MantisClone) [inline .github/workflows/build-and-test.yml]: > By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened.

Why bother excluding `reopened`?

File: conversion_proxy/Cargo.toml [hunk 8]
```diff
@@ -1,11 +1,12 @@
 [package]
 name = "conversion_proxy"
 version = "0.0.2"
-authors = ["Request Labs"]
+authors = ["Request Finance"]
```
Comment (MantisClone) [inline conversion_proxy/Cargo.toml]: Please consider adding Request Network as an author

```suggestion
authors = ["Request Finance", "Request Network"]
```

File: mocks/Cargo.toml [hunk 9]
```diff
@@ -1,11 +1,12 @@
 [package]
 name = "mocks"
 version = "0.0.2"
-authors = ["Request Labs"]
+authors = ["Request Finance"]
```
Comment (MantisClone) [inline mocks/Cargo.toml]: Please consider adding Request Network as an author.

```suggestion
authors = ["Request Finance", "Request Network"]
```

**Review by MantisClone — APPROVED** [review]
The code changes look good. :+1: Nice cleanup. 

Please consider adding Request Network to the authors list for conversion_proxy and mocks

File: tests/sim/conversion_proxy.rs (hunk 5 above)
Comment (yomarion) [reply tests/sim/conversion_proxy.rs]: Good catch, it should not. Thanks!
